crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -199,8 +168,20 @@ impl Actor for MqttActor {
     }
 
     async fn run(mut self, mut mailbox: MqttMessageBox) -> Result<(), ChannelError> {
-        while let Some(message) = mailbox.recv().await {
-            mailbox.send(message).await?;
+        loop {
+            tokio::select! {
```

Comment:
This select loop here is the reason why I had moved the MQTT connection into the message box. And my view was also slightly different. A `MessageBox` is not just a wrapper over all the peer messages but rather **all** the messages meant for an actor, irrespective of the channel they are coming from (the MQTT connection channel in this case). Both definitions can be justified in their own ways. We just need to pick the one that's more natural to the end-user.

Reply:
> We just need to pick the one that's more natural to the end-user.

I don't think so. The implementer of an actor should be free here. One key point to note, is that when I changed the MqttMessageBox, I changed neither the tests  nor the clients. This is really an implementation details that doesn't affect the users of this actor.